workflow/executor/executor.go

```diff
@@ -164,6 +164,10 @@ func (we *WorkflowExecutor) LoadArtifacts() error {
 		tempArtPath := artPath + ".tmp"
 		err = artDriver.Load(&art, tempArtPath)
 		if err != nil {
+			if art.Optional && errors.IsCode(errors.CodeNotFound, err) {
```

Comment:
Currently `Artifact.Optional` means "not specified". With this we also enhance it to mean "not found" as well. Is this what we want or should we distinguish between the two?

Some users might expect different behaviors with each of these.

Reply:
Lets discus at stand-up tomorrow, I'm having trouble getting my head around how "optional" could not be interpreted as "optional" - where as something specified in the spec can be inferred to be "not specified". It's like words have new meanings.

Reply:
```
	// Make Artifacts optional, if Artifacts doesn't generate or exist
	Optional bool `json:"optional,omitempty" protobuf:"varint,8,opt,name=optional"`
```

Hmmm....

Reply:
On further thought I agree that the two meanings are essentially the same.

"Optional" should be taken from the point of view of the container that receives the artifact. The reason as to why it's not there should not matter.